crates/trigger/src/loader.rs

```diff
@@ -204,13 +204,43 @@ fn check_uncomponentizable_module_deprecation(module_path: &Path) {
     }
 }
 
+struct TriggerComponentLoader;
```

Comment:
Are there any other component loaders?  Could we call this e.g. ComponentLoader or DefaultComponentLoader?  It doesn't seem to have anything particular to do with triggers.

Reply:
Further to the ComponentLoader comment: while labouring in the target environment mines, I realised I was going to need to call the loader/composer as part of validation.  This is needed because we validate that the _fully resolved composition,_ rather than the raw component source, targets the right world.  But I would like the validator not to depend on the `trigger` crate.

It may be that the solution for that is for the validator to implement its own ComponentLoader, and that's fine.  It's not like there's a huge amount of logic in there after all!  And we can certainly address that as a refactoring in he validation work.  Mostly just thinking out loud!

Reply:
Renamed to `ComponentLoader`.